tests/e2e/c/interchain_workflow.go

```diff
@@ -0,0 +1,160 @@
+
+		ethClient := e2e.Env.NewEthClient()
+
+		ginkgo.By("sending funds from one address to another on the C-Chain", func() {
+			// Create transaction
+			acceptedNonce, err := ethClient.AcceptedNonceAt(e2e.DefaultContext(), senderEthAddress)
+			require.NoError(err)
+			gasPrice, err := ethClient.SuggestGasPrice(e2e.DefaultContext())
+			require.NoError(err)
+			tx := types.NewTransaction(
+				acceptedNonce,
+				recipientEthAddress,
+				big.NewInt(int64(txAmount)),
+				gasLimit,
+				gasPrice,
+				nil,
+			)
+
+			// Sign transaction
+			cChainID, err := ethClient.ChainID(e2e.DefaultContext())
+			require.NoError(err)
+			signer := types.NewEIP155Signer(cChainID)
+			signedTx, err := types.SignTx(tx, signer, senderKey.ToECDSA())
+			require.NoError(err)
+
+			require.NoError(ethClient.SendTransaction(e2e.DefaultContext(), signedTx))
+
+			ginkgo.By("waiting for the C-Chain recipient address to have received the sent funds")
+			e2e.Eventually(func() bool {
+				balance, err := ethClient.BalanceAt(e2e.DefaultContext(), recipientEthAddress, nil)
+				require.NoError(err)
+				return balance.Cmp(big.NewInt(0)) > 0
+			}, e2e.DefaultTimeout, e2e.DefaultPollingInterval, "failed to see funds delivered before timeout")
+		})
+
+		// Wallet must be initialized after sending funds on the
+		// C-Chain to ensure wallet state matches on-chain state
+		ginkgo.By("initializing a keychain and associated wallet")
+		keychain := secp256k1fx.NewKeychain(senderKey, recipientKey)
+		baseWallet := e2e.Env.NewWallet(keychain)
+		xWallet := baseWallet.X()
+		cWallet := baseWallet.C()
+		pWallet := baseWallet.P()
+
+		ginkgo.By("defining common configuration")
+		avaxAssetID := xWallet.AVAXAssetID()
+		// Use the same owner for import funds to X-Chain and P-Chain
+		recipientOwner := secp256k1fx.OutputOwners{
+			Threshold: 1,
+			Addrs: []ids.ShortID{
+				recipientKey.Address(),
+			},
+		}
+		// Use the same outputs for both X-Chain and P-Chain exports
+		exportOutputs := []*secp256k1fx.TransferOutput{
+			{
+				Amt: txAmount,
+				OutputOwners: secp256k1fx.OutputOwners{
+					Threshold: 1,
+					Addrs: []ids.ShortID{
+						keychain.Keys[0].Address(),
+					},
+				},
+			},
+		}
+
+		ginkgo.By("exporting AVAX from the C-Chain to the X-Chain", func() {
+			_, err := cWallet.IssueExportTx(
+				xWallet.BlockchainID(),
+				exportOutputs,
+				e2e.WithDefaultContext(),
+			)
+			require.NoError(err)
+		})
+
+		ginkgo.By("importing AVAX from the C-Chain to the X-Chain", func() {
+			_, err := xWallet.IssueImportTx(
+				cWallet.BlockchainID(),
+				&recipientOwner,
+				e2e.WithDefaultContext(),
+			)
+			require.NoError(err)
+		})
+
+		ginkgo.By("checking that the recipient address has received imported funds on the X-Chain", func() {
+			balances, err := xWallet.Builder().GetFTBalance(common.WithCustomAddresses(set.Of(
+				recipientKey.Address(),
+			)))
+			require.NoError(err)
+			require.Greater(balances[avaxAssetID], uint64(0))
```

Comment:
totally optional - but could be:
```suggestion
			require.Positive(balances[avaxAssetID])
```